### PR TITLE
Add temp file method

### PR DESCRIPTION
**Pull request by steelbrain**
Fixes https://github.com/AtomLinter/atom-linter/issues/30

[Review on Reviewable](https://reviewable.io/reviews/atomlinter/atom-linter/39)

### CODE REVIEW

**Comment by steelbrain** [comment]
Travis is green, everything is awesome

File: lib/helpers.coffee [hunk 1]
```diff
@@ -118,3 +121,25 @@ module.exports = Helpers =
...
+    throw new Error('Invalid fileContent provided') unless typeof fileContents is 'string'
+    throw new Error('Invalid Callback provided') unless typeof callback is 'function'
+
+    return new Promise (resolve, reject) ->
+      tmp.dir {prefix: 'atom-linter_'}, (err, dirPath) ->
+        return reject(err) if err
+        filePath = path.join(dirPath, fileName)
+        fs.writeFile filePath, fileContents, (err) ->
+          if err
+            cleanupCallback()
```
Comment (SpainTrain) [inline lib/helpers.coffee]: Is this defined?

Comment (steelbrain) [inline lib/helpers.coffee]: Ouch, I removed it after I defined my own way to remove dir.

Comment (steelbrain) [inline lib/helpers.coffee]: It is the third argument with `err, dirPath`

**Pull request merged**